File: unlock.js
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

const repoDir = 'D:\\Car Rental - App';
const lockFile = path.join(repoDir, '.git', 'index.lock');

try {
  if (fs.existsSync(lockFile)) {
    fs.unlinkSync(lockFile);
    console.log('Lock file removed');
  } else {
    console.log('No lock file');
  }
} catch (e) {
  console.log('Could not remove lock:', e.message);
}

const files = [
  'app/(dashboard)/approvals/page.tsx',
  'app/(dashboard)/bookings/page.tsx',
  'app/api/approvals/route.ts',
  'app/api/bookings/[id]/route.ts',
  'app/api/bookings/route.ts',
  'components/bookings/BookingsClient.tsx',
];

try {
  const addResult = execSync(`git add ${files.map(f => `"${f}"`).join(' ')}`, { cwd: repoDir, encoding: 'utf8' });
  console.log('git add OK:', addResult);
} catch (e) {
  console.log('git add error:', e.stderr || e.message);
}

try {
  const commitResult = execSync('git commit -m "fix: remove email from profiles joins (column does not exist in profiles table)"', { cwd: repoDir, encoding: 'utf8' });
  console.log('git commit OK:', commitResult);
} catch (e) {
  console.log('git commit error:', e.stderr || e.message);
}

try {
  const pushResult = execSync('git push', { cwd: repoDir, encoding: 'utf8' });
  console.log('git push OK:', pushResult);
} catch (e) {
  console.log('git push error:', e.stderr || e.message);
}

console.log('DONE');

File: unlock.py
import os, subprocess, sys

lock = r'D:\Car Rental - App\.git\index.lock'
if os.path.exists(lock):
    os.remove(lock)
    print("Lock removed")
else:
    print("No lock file found")

os.chdir(r'D:\Car Rental - App')
files = [
    "app/(dashboard)/approvals/page.tsx",
    "app/(dashboard)/bookings/page.tsx",
    "app/api/approvals/route.ts",
    "app/api/bookings/[id]/route.ts",
    "app/api/bookings/route.ts",
    "components/bookings/BookingsClient.tsx",
]
r = subprocess.run(["git", "add"] + files, capture_output=True, text=True)
print("git add stdout:", r.stdout)
print("git add stderr:", r.stderr)

r2 = subprocess.run(["git", "commit", "-m", "fix: remove email from profiles joins (column does not exist in profiles table)"], capture_output=True, text=True)
print("git commit stdout:", r2.stdout)
print("git commit stderr:", r2.stderr)

r3 = subprocess.run(["git", "push"], capture_output=True, text=True)
print("git push stdout:", r3.stdout)
print("git push stderr:", r3.stderr)
print("DONE")
